libraries/AP_BattMonitor/AP_BattMonitor_SMBus_NeoDesign.cpp

```diff
@@ -0,0 +1,102 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include <AP_Notify/AP_Notify.h>
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_SMBus_NeoDesign.h"
+#include <utility>
+
+#define BATTMONITOR_ND_CELL_COUNT                0x5C    // cell-count register
+#define BATTMONITOR_ND_CELL_START                0x30    // first cell register
+#define BATTMONITOR_ND_CURRENT                   0x0a    // current register
+#define BATTMONITOR_ND_TEMPERATURE               0x08    // temperature register
+
+// Constructor
+AP_BattMonitor_SMBus_NeoDesign::AP_BattMonitor_SMBus_NeoDesign(AP_BattMonitor &mon,
+                                                   AP_BattMonitor::BattMonitor_State &mon_state,
+                                                   AP_BattMonitor_Params &params,
+                                                   AP_HAL::OwnPtr<AP_HAL::I2CDevice> dev)
+    : AP_BattMonitor_SMBus(mon, mon_state, params, std::move(dev))
+    , _cell_count(0)
```

Comment:
This is unneeded, we will start with all 0'd memory, so we try and avoid things like this in constructors as it just burns flash. The same is true for the `_pec_supported` line below. Although that one is serving more as a doc line at the moment. (Really though we want PEC support...)